### PR TITLE
fix: align gen1 substitute status behavior

### LINKED ISSUE #908: bug: gen1 docs and tests contain stale contradictory claims about move inventory and implemented mechanics
## Description
Several `gen1` docs/tests still describe package state that no longer matches the shipped package. The most concrete drift is the move inventory count: the shipped data and assertions are 165 moves, but the README, status doc, and at least one test title still say 164. There are also stale `it.todo(...)` entries that claim mechanics are not implemented even though other current code/tests prove they are.

## Severity
LOW

## Package / Location
- `packages/gen1/README.md:9`
- `specs/reference/gen1-status.md:133`
- `packages/gen1/tests/data-loading.test.ts:254-260`
- `packages/gen1/tests/gen1-mechanics.test.ts:382-384`
- `packages/gen1/tests/ruleset-branches.test.ts:1927-1929`
- `packages/gen1/tests/bug-sweep-fixes.test.ts:836-858`

## Expected vs Actual
- Expected: README, status docs, and test descriptions should describe the same package state the shipped data and passing assertions describe.
- Actual:
  - README says `164 moves`
  - `gen1-status.md` says `164 moves loaded`
  - `data-loading.test.ts` title says `contains 164 moves` while the assertion expects `165`
  - stale TODOs claim Reflect screen handling and Hyper Beam KO recharge-skip are not implemented even though current code/tests already cover them

## Why It Matters
This creates false signals during audits and maintenance work. Engineers reading the docs/tests can conclude the package still has 164 moves or that mechanics are missing when they are already implemented. That increases review noise and makes test coverage harder to trust.

## Evidence
1. Move-count drift:
   - `packages/gen1/README.md` line 9: `**164 moves**`
   - `specs/reference/gen1-status.md` line 133: `151 species, 164 moves loaded`
   - `packages/gen1/tests/data-loading.test.ts` line 254: test title says `contains 164 moves`
   - same test asserts `expect(allMoves.length).toBe(165)` at line 260
   - direct data check: `jq 'length' packages/gen1/data/moves.json` returns `165`
2. Stale TODO drift:
   - `packages/gen1/tests/gen1-mechanics.test.ts` TODO says Reflect damage-calc screen effect is "not yet in calculateGen1Damage"
   - `packages/gen1/src/Gen1DamageCalc.ts` currently applies Reflect / Light Screen defense doubling
   - `packages/gen1/tests/ruleset-branches.test.ts` TODO says Hyper Beam KO recharge skip is "not yet implemented in engine"
   - `packages/gen1/tests/bug-sweep-fixes.test.ts` already has passing assertions that `noRecharge` is true on KO

## Reproduction Notes
- No code changes required.
- The contradictions are visible by reading the listed files and by checking `moves.json` length.

## Source References Used To Verify
- `packages/gen1/README.md`
- `specs/reference/gen1-status.md`
- `packages/gen1/tests/data-loading.test.ts`
- `packages/gen1/tests/gen1-mechanics.test.ts`
- `packages/gen1/tests/ruleset-branches.test.ts`
- `packages/gen1/tests/bug-sweep-fixes.test.ts`
- `packages/gen1/src/Gen1DamageCalc.ts`
- `packages/gen1/data/moves.json`

## Impact / Risk
- Misleads consumers and contributors about the shipped data inventory.
- Reduces trust in tests as documentation.
- Creates avoidable audit/review churn because stale TODOs look like open implementation gaps.

## Suggested Fix Direction
Synchronize README/status/test descriptions with the current 165-move data set, and remove or update stale TODOs/comments whose mechanics are already implemented and covered elsewhere.

## How Found
Found during an exhaustive package audit of `packages/gen1` on 2026-03-23.

---
Filed by Codex agent

### LINKED ISSUE #915: testing: gen1 Substitute status tests only assert fixture flags and conflict with ground-truth notes
## Summary
`packages/gen1/tests/gen12-edge-case-interactions.test.ts` contains four tests named as Substitute runtime-behavior checks, but they only assert `bypassSubstitute: false` on local fixture objects. They never execute the engine or ruleset behavior they describe, and their comments contradict the current Gen 1 ground-truth doc for at least sleep/paralysis through Substitute.

## Severity
Low

## Package / Location
- Package: `@pokemon-lib-ts/gen1`
- File: `packages/gen1/tests/gen12-edge-case-interactions.test.ts`

## Expected Behavior
Tests with names like “Sleep Powder cannot bypass Substitute” should exercise the actual runtime path and assert the real Gen 1 behavior.

If the repo’s current Gen 1 source hierarchy is followed, tests should also reflect that Substitute does **not** block sleep/paralysis from status moves.

## Actual Behavior
The four tests under `describe("Gen 1 Substitute blocks status moves")` only construct move objects and assert `expect(move.flags.bypassSubstitute).toBe(false)`. They do not create a battle state, do not call the engine, and do not observe Substitute behavior at all.

The surrounding comments also claim that sleep/paralysis/confusion status moves “all fail against a Substitute”, which conflicts with `specs/reference/gen1-ground-truth.md`.

## Why It Matters
These tests create false confidence. They read like mechanic-regression coverage, but they prove nothing about runtime Substitute handling. Because the comments also contradict the current ground-truth doc, they can mislead future refactors in the wrong direction.

## Exact Evidence
Test source:
- `packages/gen1/tests/gen12-edge-case-interactions.test.ts`
  - describe block: `Gen 1 Substitute blocks status moves`
  - each test only checks `move.flags.bypassSubstitute === false`

Authoritative local reference:
- `specs/reference/gen1-ground-truth.md`
  - `Substitute` section states: `Does NOT block: sleep, paralysis from status moves`

Runtime proof sketch:
- Running a real Gen 1 battle with an active Substitute and `Sleep Powder` shows the target can still become asleep while Substitute remains active.
- That means these tests are not just weak; their comments point away from the currently documented behavior.

## Reproduction Notes
Example runtime check used during audit:
- Turn 1: defender uses `Substitute`
- Turn 2: attacker uses `Sleep Powder` into the active substitute
- Observed: target status becomes `sleep`

## Impact / Risk
- low-value regression coverage
- misleading test names/comments
- future contributors may “fix” runtime toward an incorrect oracle

## Suggested Fix Direction
Replace these fixture-flag checks with real runtime tests against the engine/ruleset path, and align the assertions/comments with the repo’s current Gen 1 source hierarchy.

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary
- align Gen 1 Substitute status resolution with the local ground-truth exceptions for sleep and paralysis
- replace misleading flag-only Substitute tests with ruleset-path behavior assertions
- fix the remaining Gen 1 move-count doc drift to 165 moves

## Verification
- `npx vitest run packages/gen1/tests/unit/gen12-edge-case-interactions.test.ts packages/gen1/tests/unit/gen1-mechanics.test.ts packages/gen1/tests/unit/formula-bug-fixes.test.ts`
- `npm run test --workspace @pokemon-lib-ts/gen1`
- `npm run typecheck --workspace @pokemon-lib-ts/gen1`
- `npx @biomejs/biome check packages/gen1/src/Gen1Ruleset.ts packages/gen1/tests/unit/gen12-edge-case-interactions.test.ts packages/gen1/README.md specs/reference/gen1-status.md`

Closes #908
Closes #915

## Summary by CodeRabbit

* **Bug Fixes**
  * Refined Substitute behavior: it now selectively blocks secondary effects, volatile inflictions, and some stat changes, while sleep and paralysis from status moves still apply.

* **Documentation**
  * Updated Gen 1 reference: move count increased from 164 to 165.

* **Tests**
  * Revised tests to match the tightened Substitute interaction behavior.

### CODE REVIEW

File: packages/gen1/src/Gen1Ruleset.ts [hunk 1]
```diff
@@ -1188,6 +1205,22 @@ export class Gen1Ruleset implements GenerationRuleset {
...
+    context: MoveEffectContext,
+    move: MoveData,
+    status?: PrimaryStatus,
+  ): boolean {
+    if (context.defender.substituteHp <= 0 || move.flags.bypassSubstitute) {
+      return false;
+    }
+
+    return status !== CORE_STATUS_IDS.sleep && status !== CORE_STATUS_IDS.paralysis;
+  }
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen1/src/Gen1Ruleset.ts]: Addressed in b4adc58e and retained on the current head: `isBlockedBySubstitute` now treats `context.brokeSubstitute` as blocked even after `substituteHp` reaches 0, and `gen12-edge-case-interactions` now covers the substitute-breaking secondary-effect path explicitly.
